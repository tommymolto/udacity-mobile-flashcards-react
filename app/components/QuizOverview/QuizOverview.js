import React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components/native';
import PropTypes from 'prop-types';
import ButtonWrapper from '../Buttons/ButtonWrapper';
import PrimaryButton from '../Buttons/PrimaryButton';
import { resetQuiz } from '../../actions/quizActions';

const StyledView = styled.View`
  flex: 1;
`;

const StyledTitleText = styled.Text`
  font-size: 20;
  padding-vertical: 15;
  padding-horizontal: 15;
  text-align: center;
`;

const StyledQuestionsNumText = styled.Text`
  font-size: 16;
  font-style: italic;
  padding-vertical: 15;
  padding-horizontal: 15;
  text-align: center;
`;

const QuizOverview = (props) => {
  const { title, questions } = props.navigation.state.params;
  const volta = () => {
    props.navigation.navigate('QuizOverview', {
      title,
      questions,
    });
  };
  return (
    <StyledView>
      <StyledTitleText>Quiz: {title}</StyledTitleText>
      <StyledQuestionsNumText>{props.quizLength} cards</StyledQuestionsNumText>
      <ButtonWrapper>
        <PrimaryButton
          text="Start quiz"
          onPress={() => {
            props.userResetQuiz();
            // console.log('aqui');
            props.navigation.navigate('Quiz', {
              title,
              questions,
              volta,
            });
          }
          }
        />
        <PrimaryButton
          text="Add new question"
          onPress={() => props.navigation.navigate('NewCardForm', { title })}
        />
        <PrimaryButton
          text="Home"
          onPress={() => props.navigation.navigate('Home')}
        />
      </ButtonWrapper>
    </StyledView>
  );
};

QuizOverview.propTypes = {
  navigation: PropTypes.object.isRequired,
  quizLength: PropTypes.number.isRequired,
  volta: PropTypes.func.isRequired,
  userResetQuiz: PropTypes.func.isRequired,
};

const mapStateToProps = (state, ownProps) => ({
  quizLength: state.decks[ownProps.navigation.state.params.title].quizLength,
  volta: () => {
    this.props.navigation.navigate('QuizOverview', props)
  },
});
const mapDispatchToProps = dispatch => ({
  userResetQuiz: () => {
    dispatch(resetQuiz());
  },
});


export default connect(mapStateToProps, mapDispatchToProps)(QuizOverview);
