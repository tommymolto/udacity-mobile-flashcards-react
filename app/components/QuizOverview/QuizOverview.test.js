/* global describe, it, expect */

import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import QuizOverview from './QuizOverview';
import storeConfig from '../../store/storeConfig';

describe('QuizOverview', () => {
  it('renders correctly', () => {
    const navigation = {
      state: {
        params: {
          title: 'React',
          questions: 2,
        },
      },
    };
    const tree = renderer

      .create(<Provider store={storeConfig}>
        <QuizOverview navigation={navigation} />
              </Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
