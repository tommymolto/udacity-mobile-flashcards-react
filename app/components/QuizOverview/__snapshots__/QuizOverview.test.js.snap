// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizOverview renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 20,
          "paddingHorizontal": 15,
          "paddingVertical": 15,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    Quiz: 
    React
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 16,
          "fontStyle": "italic",
          "paddingHorizontal": 15,
          "paddingVertical": 15,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    2
     cards
  </Text>
  <View
    style={
      Array [
        Object {
          "marginVertical": 50,
          "paddingBottom": 25,
          "paddingHorizontal": 25,
          "paddingTop": 20,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "#05AE54",
              "color": "#ffffff",
              "fontSize": 22,
              "height": 45,
              "marginLeft": 40,
              "marginRight": 40,
              "marginVertical": 10,
              "paddingBottom": 10,
              "paddingTop": 10,
              "textAlign": "center",
            },
            undefined,
          ]
        }
      >
        Start quiz
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "#05AE54",
              "color": "#ffffff",
              "fontSize": 22,
              "height": 45,
              "marginLeft": 40,
              "marginRight": 40,
              "marginVertical": 10,
              "paddingBottom": 10,
              "paddingTop": 10,
              "textAlign": "center",
            },
            undefined,
          ]
        }
      >
        Add new question
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "#05AE54",
              "color": "#ffffff",
              "fontSize": 22,
              "height": 45,
              "marginLeft": 40,
              "marginRight": 40,
              "marginVertical": 10,
              "paddingBottom": 10,
              "paddingTop": 10,
              "textAlign": "center",
            },
            undefined,
          ]
        }
      >
        Home
      </Text>
    </View>
  </View>
</View>
`;
